### PR TITLE
Broken poly

### PR DESCRIPTION
**Pull request by viktorya-grigoryan**
## Description
Broken polynomial
## Linked Issue
Fixes # (issue number)

## Implementation Checklist
- [x] My class inherits from `PolynomialBasis` (if applicable).
- [x] I have implemented all abstract methods (`evaluate_basis`, `fit`, etc.).
- [x] I have used `numpy` vectorization to avoid inefficient Python loops.
- [x] Documentation strings (docstrings) are provided for all public methods.

### 1. Mathematical Integrity
- [ ] **Convergence:** Does the implementation maintain expected accuracy? (Checked for Runge's phenomenon or stability).
- [ ] **Edge Cases:** Does it handle single points or non-equally spaced nodes correctly?
- [ ] **Domain Mapping:** Does the class correctly map the input domain to the basis domain (e.g., [-1, 1] for Chebyshev)?

### 2. API & Style
- [ ] **Method Signatures:** Methods match the Abstract Base Class exactly.
- [ ] **Naming:** Follows `CamelCase` for classes and `snake_case` for functions/variables.
- [ ] **Clean Code:** No "magic numbers"; constants are named and explained.

### 3. Testing Quality
- [ ] **Coverage:** Unit tests cover typical usage and edge cases.
- [ ] **Precision:** Tests use `np.testing.assert_allclose` instead of `==`.
- [ ] **Validation:** There is a test proving the basis can exactly reconstruct a polynomial of its own degree.

## Screenshots / Plots

### CODE REVIEW

File: tests/test_broken.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    x_right = 1.0 + 1e-6
+
+    y_left = basis.evaluate(coeffs, np.array([x_left]))[0]
+    y_right = basis.evaluate(coeffs, np.array([x_right]))[0]
+
+    assert y_left != y_right
+
+
+
+# 6. Test coefficient structure
```
Comment (mtezzele) [inline tests/test_broken.py]: remove unnecessary comments before the methods please

File: tests/test_broken.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+def test_broken_coeff_structure():
+
```
Comment (mtezzele) [inline tests/test_broken.py]: remove empty lines between the method and the doc

File: src/numanalysislib/basis/broken.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        Each element has its own independent polynomial basis.
+        No continuity is enforced between elements.
+
+        DOFs = N_elements * (degree + 1)
+        """
+        self.basis_type = basis_type
+
+        # Build mesh as list of intervals
+        # xmpl: [0,1,2] → [(0,1), (1,2)]
+        self.mesh = [(mesh[i], mesh[i+1]) for i in range(len(mesh)-1)]
```
Comment (xc900606) [inline src/numanalysislib/basis/broken.py:32]: It is better to validate the mesh in case of repeated or unsorted mesh points that create zero length or reversed elements.

File: src/numanalysislib/basis/broken.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+        Global index → (element, local index)
+        """
+        x = np.asarray(x)
+
+        element_id = index // self.local_dofs
+        local_index = index % self.local_dofs
+
+        element = self.mesh[element_id]
+        a, b = element
```
Comment (xc900606) [inline src/numanalysislib/basis/broken.py:71]: Should we also validate the index here and reject negative or too-large values?

File: src/numanalysislib/basis/broken.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+            xq = 0.5*(b - a)*xi + 0.5*(a + b)
+            wq = 0.5*(b - a)*wi
+
+            # Build mass matrix M_ij = ∫ φ_i φ_j
+            M = np.zeros((n, n))
+            for i in range(n):
+                phi_i = basis.evaluate_basis(i, xq)
+                for j in range(n):
+                    phi_j = basis.evaluate_basis(j, xq)
+                    M[i, j] = np.sum(wq * phi_i * phi_j)
```
Comment (xc900606) [inline src/numanalysislib/basis/broken.py]: This recomputes basis values inside a nested loop. Would it make sense to compute them once and reuse them when building M?

File: src/numanalysislib/basis/broken.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+                return self.bases[element].evaluate(coefficients[element], np.array([x]))[0]
+
+        return 0.0  # outside domain
+
+    def evaluate(self, coefficients: dict, x: np.ndarray) -> np.ndarray:
+        """
+        Vectorized evaluation
+        """
+        x = np.asarray(x)
+        return np.vectorize(lambda xi: self.float_evaluate(coefficients, xi))(x)
```
Comment (xc900606) [inline src/numanalysislib/basis/broken.py]: np.vectorize still evaluates xi one point at a time, which can be inefficient.

**Review by xc900606 — COMMENTED** [review]
I left a few comments, mostly about edge cases, especially mesh validation and clearer handling of invalid basis indices. The core implementation and the main tests look good overall.

File: src/numanalysislib/basis/broken.py (hunk 6 above)
Comment (mtezzele) [reply src/numanalysislib/basis/broken.py]: the fastest fix to this is to use:
`np.array([self.float_evaluate(coefficients, xi) for xi in x])`

**Comment by viktorya-grigoryan** [comment]
Fixed all the comments
